bin/download-translations.js

```diff
@@ -0,0 +1,178 @@
+			);
+		});
+	},
+
+	/**
+	 * Download given language sets for formats listed in the config (mo/po).
+	 *
+	 * @param {array} sets The translation sets from the GlotPress server.
+	 */
+	downloadAll: function( sets ) {
+		sets.forEach( set => {
+			options.formats.forEach( format => {
+				translations.downloadSet( set, format );
+			});
+		});
+	},
+
+	/**
+	 * Download the translation file for the given translation set and format.
+	 *
+	 * Example download: https://translate.studiopress.com/global/api/projects/genesis/genesis/en-gb/default/export-translations?format=po
+	 *
+	 * @param {object} set Information returned by the GlotPress server for an
+	 *                     individual set, such as 'en_GB'.
+	 * @param {string} format The translation format, such as 'mo' or 'po'.
+	 */
+	downloadSet: function( set, format ) {
+		const url =
+			translations.apiUrl +
+			'/' +
+			set.locale +
+			'/' +
+			set.slug +
+			'/export-translations?format=' +
+			format;
+
+		const info = {
+			domainPath: options.domainPath || '',
+			textdomain: options.textdomain,
+			locale: set.locale,
+			wpLocale: set.wp_locale || set.locale,
+			slug: set.slug,
+			slugSuffix: 'default' === set.slug ? '' : '-' + set.slug,
+			format: format
+		};
+
+		translations.getFile( url, translations.buildFilename( info ) );
+	},
+
+	/**
+	 * Get the local file path and name to use for the downloaded file.
+	 *
+	 * Substitutes placeholders wrapped in `#` in the `glotpress.fileFormat`
+	 * value in `package.json` with properties from the supplied setInfo
+	 * object. For example, `lib/languages/%wpLocale%.%format%"` might become
+	 * `lib/languages/en-GB.mo"`.
+	 *
+	 * @param {object} setInfo Information about the translation set.
+	 * @return {string} The local filename relative to the project root.
+	 */
+	buildFilename: function( setInfo ) {
+		return options.fileFormat.replace( /%(\w*)%/g, ( _, key ) => {
+			return setInfo.hasOwnProperty( key ) ? setInfo[key] : '';
+		});
+	},
+
+	/**
+	 * Get a file from a given URL and save it with the provided filename.
+	 *
+	 * @param {string} url The URL to download.
+	 * @param {string} filename The file path and name to write to,
+	 *                          relative to the project root.
+	 */
+	getFile: function( url, filename ) {
+		axios({
+			method: 'get',
+			url: url,
+			responseType: 'arraybuffer'
+		}).then( response => {
+			if ( 200 === response.status ) {
+				const encoding = filename.includes( '.mo' ) ? 'binary' : 'utf8';
+				fs.writeFile(
+					process.cwd() + '/' + filename,
+					response.data,
+					encoding,
+					err => {
+						if ( err ) {
+							console.log( `Could not write language file: ${err}` );
+							return;
+						}
+						console.log( `Writing language file ${filename}` );
+					}
+				);
+			} else {
+				console.log(
+					`Could not fetch file. Response: ${response.status} ${response.statusText}`
+				);
+			}
+		});
+	}
+};
+
+translations.init();
```

Comment:
May be redundant for a script like this, but should / could this `translations` be exported, and the `translations.init()` be called in another file? Thinking of code being defined alongside side-effects.

Reply:
If this was a package, yes, I agree. We would:

1. Expose the init function with a global name like `downloadTranslations` (or `glotpress` if we planned to extend functionality beyond only downloads).
2. Host the package on npm or GitHub (GitHub now has a [package registry](https://help.github.com/en/articles/about-github-package-registry)).
3. Include the package in dev dependencies.
4. Run the script by name in npm scripts (`downloadTranslations --optional-flags-here`).

We're using `download-tranlsations.js` as an alternative to a shell script that better fits with the other tooling and running it directly only via npm scripts, though. I don't think a refactor that follows the above steps is required here. If we want to make the translation script available more widely later, we could certainly consider it then, though.